src/main/java/hudson/plugins/gearman/GearmanProxy.java

```diff
@@ -71,18 +72,19 @@ private GearmanProxy() {
         // query Jenkins for built-in name
         try {
             builtInNode = Jenkins.get().getComputer("");
-            hostname = builtInNode.getHostName();
+            if (builtInNode != null) {
+                hostname = builtInNode.getHostName();
+            } else {
+                // Built-in node may not be enabled so get builtInName from system
+                try {
+                    hostname = java.net.InetAddress.getLocalHost().getHostName();
+                } catch (UnknownHostException e) {
+                    logger.warn("Exception while getting built-in node hostname", e);
+                }
+            }
         } catch (Exception e) {
             logger.warn("Exception while getting hostname", e);
```

Comment:
The code seems to be perfectly valid except that the thread naming will be imperfect. I think logging at INFO level would make more sense.